src/NosCore.GameObject/ComponentEntities/Interfaces/ICharacterEntity.cs

```diff
@@ -71,5 +71,11 @@ public interface ICharacterEntity : INamedEntity, IExperiencedEntity
         ConcurrentDictionary<Guid, CharacterRelation> RelationWithCharacter { get; }
 
         ConcurrentDictionary<long, long> GroupRequestCharacterIds { get; }
+
+        void SetLevel(byte level);
```

Comment:
not link to Character ;) it's for the Interface that got a level (IExperiencedEntity maybe) in this case move the SetLevel to the extension (our pet can level up too)